### ISSUE TITLE
Scalers crash when run locally on Macs but not on PCs

### ISSUE THREAD

**Issue by yoDon** [microsoft/azure-container-apps#406]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is a:
- [X] documentation issue or request

### Issue description
This issue is a documentation suggestion, but I'm not sure where to add it so I wasn't able to just submit a PR.

When running an ACA project locally under OSX for development and testing purposes, Scalers like the one in the [Azure Container Apps Orleans sample](https://github.com/Azure-Samples/Orleans-Cluster-on-Azure-Container-Apps)) crash while reporting:
```
"HTTP/2 over TLS is not supported on macOS due to missing ALPN support.
```
The Silo itself runs fine, and the various minimal web api projects that interact with the silo also work fine, but the Scaler wont run on macOS. The Scaler project is crashing because the gRPC KEDA service it hosts is trying to bind to a port using HTTP/2 over TLS which is not possible on a Mac (and also not possible on Windows 7 or earlier). This exception does not get thrown when running the same solution under current versions of Windows.

The solution to the issue is to add the following code to `Scaler/program.cs`:
```
//
// NOTE - ONLY USE THIS CODE FOR LOCAL DEVELOPMENT
// NOTE - DO NOT USE THIS CODE TO TURN OFF TLS IN PRODUCTION
// NOTE - This code is only needed on Macs and on Windows 7 or earlier
//
builder.WebHost.ConfigureKestrel(options =>
{
    // Setup a HTTP/2 endpoint without TLS.
    options.ListenLocalhost(7122, o => o.Protocols =
        HttpProtocols.Http2);
});
```

By way of additional background, TLS issues on macOS seem to be a well-known aspnetcore 6 issue, discussed for example [here](https://learn.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?view=aspnetcore-6.0#unable-to-start-aspnet-core-grpc-app-on-macos). 

### Steps to reproduce
1. Use a Mac
2. Clone https://github.com/Azure-Samples/Orleans-Cluster-on-Azure-Container-Apps.git
3. Edit one line in `Scaler/program.cs` to use `.UseLocalhostClustering();` instead of `.UseAzureStorageClustering(...);` (so you can run that example locally rather than in the cloud)
4. Set the Scaler as the startup project for the solution (you only need to run the Scaler to see this issue manifest)
5. Run the debugger

**Expected behavior** [What you expected to happen.]
1. Project runs without error

**Actual behavior** [What actually happened.]
1. An exception popup reports
```
System.IO.IOException has been thrown
Failed to bind to address https://localhost:7122
```
2. Click on Details in the error popup and drill down to the nested `System.NotSupportedException`
3. See that the Message property of the NotSupportedException is `HTTP/2 over TLS is not supported on macOS due to missing ALPN support.`

**Comment by kendallroden** [comment]
Hi- this seems to be an issue with a specific code base stored in a different repository. This is unrelated to the container apps service and appears to be a note related to the repo cc: @bradygaster I recommend opening the issue on that repository @yoDon

**Comment by bradygaster** [comment]
Agreed with Kendall - can you file this issue in that sample's repository rather than on the service team?

**Comment by bradygaster** [comment]
Also, take a look at [this documentation page](https://learn.microsoft.com/en-us/aspnet/core/tutorials/grpc/grpc-start?view=aspnetcore-6.0&tabs=visual-studio), as I think it summarizes the crux of the issue you're having and how to work around it. the TL/DR is this:

![image](https://user-images.githubusercontent.com/41929050/194900848-f25cea96-a067-4dc6-8f59-c9d9137b785b.png)

**Comment by yoDon** [comment]
It's of course your call, but given the criticality of gRPC-based scalars to ACA projects and the cross-platform nature of the local dev-and-test story for ACA, it feels to me like there should be a tracked issue for this at the platform level, even if there's not currently an approach you can see to finding a fix. 

Again, it's your call, but it's hard for me to see how the fix should be the responsibility of some random sample I happened to pick that happened to exhibit this issue so there'd be an easy repro in the issue report. To me this feels like a hard-to-fix ACA platform local dev-and-test issue.

**Comment by bradygaster** [comment]
@yoDon - we do have an issue tracking this part of the experience over in the ASP.NET Core repo. Probably good to track progress on this part of the experience there, and to close this issue as, again, it has little-to-nothing to do with the ACA team. I agree, it's a development experience opportunity for .NET developers using Macs to build their apps, but that's not the responsibility of the ACA team to fix it. :) 

https://github.com/dotnet/runtime/issues/27727
